### PR TITLE
[Test Proxy] Update recording asset transition script links

### PR DESCRIPTION
**Pull request by mccoyp**
Fixes outdated links to asset sync documentation.

### CODE REVIEW

**Review by scbedd — APPROVED** [review]
❤️

File: tools/test-proxy/documentation/asset-sync/README.md [hunk 1]
```diff
@@ -153,7 +153,7 @@ First, ensure that your language-specific "shim" supports the automatic addition
 - [PR Enabling in JS](https://github.com/Azure/azure-sdk-for-js/pull/23405)
 - [PR Enabling in Go](https://github.com/Azure/azure-sdk-for-go/pull/19322)
 
-Use [the transition script](../../scripts/transition-scripts/generate-assets-json.ps1) and follow the [readme](../../scripts/transition-scripts/README.md)!
+Use [the transition script](../../../../eng/common/testproxy/transition-scripts/generate-assets-json.ps1) and follow the [readme](../../../../eng/common/testproxy/transition-scripts/README.md)!
```
Comment (benbp) [inline tools/test-proxy/documentation/asset-sync/README.md:156]: Should we just use absolute links here so if this file location changes the links don't break?

Comment (mccoyp) [reply tools/test-proxy/documentation/asset-sync/README.md:156]: That's a good point; I stuck to relative links to preserve the style, but absolute links would be more resilient

Comment (benbp) [reply tools/test-proxy/documentation/asset-sync/README.md]: Also using absolute links lets our tooling do link validation.

**Pull request merged**